src/main/java/org/folio/circulation/domain/RequestQueue.java

```diff
@@ -86,6 +85,8 @@ public Boolean contains(Request request) {
   Collection<Request> getRequestsWithChangedPosition() {
     return requests.stream()
       .filter(Request::hasChangedPosition)
+      // order by position descending
+      .sorted((req1, req2) -> req2.getPosition().compareTo(req1.getPosition()))
```

Comment:
Does this mean the approach is to change the request in the lowest position (highest number) first?

Will that work when the same logic is used during request cancellation, and the request in position 9 is moving to position 8?

Reply:
The `getRequestsWithChangedPosition` is only called before updating the requests in which order changed. The add or remove of a requests performs a reordering that is expected. Therefor, when a request is cancelled it is removed and reorders accordingly. FYI, the tests pass and if covered no false positive occurs it works as expected.

I am going to slightly redesign the reordering and updating request positions to accommodate more complex scenarios.

Reply:
> The `getRequestsWithChangedPosition` is only called before updating the requests in which order changed. The add or remove of a requests performs a reordering that is expected. Therefor, when a request is cancelled it is removed and reorders accordingly. 

I don't think I follow. Is `getRequestsWithChangedPosition` only used when a request is added to the queue, and something else is used when a request is removed?

> FYI, the tests pass and if covered no false positive occurs it works as expected.

Given that the constraint can affect the other processes, like cancellation as well, too. The false positives you refer to are possible in those circumstances as well. Which means I don't think we can assume that the new implementation doesn't hit the constraint under any circumstances.

Reply:
`RequestQueue.add` and `RequestQueue.remove` both reorder and update the positions. `getRequestsWithChangedPosition` is only called from `updateRequestsWithChangedPositions`, which is called is several places. Including `onCancellation`, `onMoveFrom`, `onMoveTo`, etc.